### PR TITLE
Cmd top

### PR DESCRIPTION
**Pull request by AdriannaPienkowska**

### CODE REVIEW

**Comment by Ernold11** [comment]
Check List
----------
* [ ]  CI is passed
* [ ]  master is merged into current branch
* [ ]  public methods should be at the top of class, protected at bottom  
       example: `on_new_line()` is public, `_parse()` is protected
* [ ]  Name of methods should tell what its doing, not how are doing  
       example: `_parse_re_via_dev_proto_metric()` -> `_parse_via_dev_proto_metric()`
* [ ]  All parsing lines (on_new_line) should be covered by unit tests
* [ ]  Be sure that  COMMAND_OUTPUT_* / COMMAND_KWARGS_* / COMMAND_RESULT_* cover all parsing lines
* [ ]  Parsed line example, parsing regexp, parsing method should be next to each other
* [ ]  Using named groups inside regexp (like `r"^\s*(?P<ADDRESS>\S+)\s+via\s+(?P<VIA>\S+)”`) is preferred
* [ ]  Longer regexp are before shorter regexp
* [ ]  Resources for pytest should be declared as `@pytest.fixture` or `@pytest.yield_fixture`
* [ ]  There are no sensitive data (company specific) like: IP addr, passwords, commands, tools etc

**Comment by Ernold11** [comment]
Please increase coverage into 100%

File: moler/cmd/unix/top.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+__email__ = 'adrianna.pienkowska@nokia.com'
+
+from moler.cmd.unix.genericunix import GenericUnixCommand
+from moler.exceptions import CommandFailure
+from moler.exceptions import ParsingDone
+import re
+
+
+class Top(GenericUnixCommand):
+    def __init__(self, connection, options=None):
```
Comment (Ernold11) [inline moler/cmd/unix/top.py]: Add prompt and newline chars

File: moler/cmd/unix/top.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        self.current_ret.update({mem_type: mem_row_dict})
+
+    def _parse_processes_list(self, line):
+        if self._processes_list_headers:
+            processes_info = line.strip().split()
+            processes_dict = dict(zip(self._processes_list_headers, processes_info))
+            self.current_ret['processes'].append(processes_dict)
+
+
+COMMAND_OUTPUT_without_options = """
```
Comment (Ernold11) [inline moler/cmd/unix/top.py]: this is not output from `top` command. Top command is infinity running command.

File: moler/cmd/unix/top.py [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+        super(Top, self).__init__(connection=connection)
+        self.options = options
+        self._processes_list_headers = list()
+        self.current_ret = dict()
+
+    def build_command_string(self):
+        cmd = "top"
+        if self.options:
+            cmd = cmd + " " + self.options + " n 1"
+        return cmd
```
Comment (Ernold11) [inline moler/cmd/unix/top.py]: If options then only run command once, otherwise infinity loop?

File: moler/cmd/unix/top.py [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    _re_cpu_row = re.compile(r'.*(?P<CPU_ROW>Cpu).*:\s*(?P<US>\d*.\d*).*us,\s*(?P<SY>\d*.\d*).*sy,\s*(?P<NI>\d*.\d*)'
+                             r'.*ni,\s*(?P<ID>\d*.\d*).*id,\s*(?P<WA>\d*.\d*).*wa,\s*(?P<HI>\d*.\d*)'
+                             r'.*hi,\s*(?P<SI>\d*.\d*).*si,\s*(?P<ST>\d*.\d*).*st')
+
+    _re_memory_rows = re.compile(r'(?P<MEM>.*):\s*(?P<TOTAL_MEM>\d*)(?P<UNIT>.)\s*total,\s*(?P<FREE>\d*)\s*free,\s*'
+                                 r'(?P<USED>\d*)\s*used[,.]\s*(?P<OTHER>\d*)\s*')
+
+    _re_processes_header = re.compile(r'(?P<HEADER> .*PID.*)')
+
+    def _parse_line(self, line):
```
Comment (Ernold11) [inline moler/cmd/unix/top.py]: single if in single method

File: moler/cmd/unix/top.py [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+ 11447 bylica    20   0  325000  20772  14380 S  1.3  1.0   0:38.90 lxterminal
+  9497 root      20   0   46796   3844   3068 R  0.3  0.2   0:00.91 top
+     1 root      20   0  138888   4500   3316 S  0.0  0.2   0:01.93 systemd
+xyz@debian:~$"""
+
+COMMAND_KWARGS_without_options = {
+
+}
+
+COMMAND_RESULT_without_options = {
```
Comment (Ernold11) [inline moler/cmd/unix/top.py]: Number values should be cast into int not string

**Comment by Ernold11** [comment]
Still not 100% coverage.
When you finish make necessary changes please write comment into pull request (ready to review)

File: moler/cmd/unix/top.py [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+                             r'.*ni,\s*(?P<ID>\d*.\d*).*id,\s*(?P<WA>\d*.\d*).*wa,\s*(?P<HI>\d*.\d*)'
+                             r'.*hi,\s*(?P<SI>\d*.\d*).*si,\s*(?P<ST>\d*.\d*).*st')
+
+    _re_memory_rows = re.compile(r'(?P<MEM>.*):\s*(?P<TOTAL_MEM>\d*)(?P<UNIT>.)\s*total,\s*(?P<FREE>\d*)\s*free,\s*'
+                                 r'(?P<USED>\d*)\s*used[,.]\s*(?P<OTHER>\d*)\s*')
+
+    _re_processes_header = re.compile(r'(?P<HEADER> .*PID.*)')
+
+    def _parse_line(self, line):
+        if self._regex_helper.search_compiled(Top._re_top_row, line):
```
Comment (Ernold11) [inline moler/cmd/unix/top.py]: Please use common template for parsing in on_new_line.
Move parsing method into on_new_line.
Move checking regex into parsing method.

**Comment by AdriannaPienkowska** [comment]
Ready for review

**Pull request merged**